src/components/views/dialogs/ErrorDialog.js

```diff
@@ -51,22 +51,18 @@ export default React.createClass({
         };
     },
 
-    componentDidMount: function() {
-        if (this.props.focus) {
-            this.refs.button.focus();
-        }
-    },
-
     render: function() {
         const BaseDialog = sdk.getComponent('views.dialogs.BaseDialog');
         return (
             <BaseDialog className="mx_ErrorDialog" onFinished={this.props.onFinished}
-                    title={this.props.title || _t('Error')}>
-                <div className="mx_Dialog_content">
+                    title={this.props.title || _t('Error')}
+                    contentId='mx_Dialog_content'
+            >
+                <div className="mx_Dialog_content" id='mx_Dialog_content'>
                     { this.props.description || _t('An error has occurred.') }
                 </div>
                 <div className="mx_Dialog_buttons">
-                    <button ref="button" className="mx_Dialog_primary" onClick={this.props.onFinished}>
+                    <button className="mx_Dialog_primary" onClick={this.props.onFinished} autoFocus={this.props.focus}>
```

Comment:
I'm not sure where `focus` is coming from. Should `autoFocus` not always be `true`?

Reply:
Error dialog has focus prop that is true by default. In fact I haven't introduced it my-self. It used button's ref to set the focus conditionally in the componentDidMount function and I have changed it to use autoFocus prop of the button component instead as I like it better. This is just cosmetic change and the functionality should not be affected.

Reply:
My mistake, ErrorDialog indeed has this property defined - I was obviously looking in the from file somehow. 

Thanks for changing that - `autoFocus` is a clear improvement over using refs.